x-pack/test/api_integration/apis/ml/jobs/stop_datafeeds.ts

```diff
@@ -0,0 +1,245 @@
+      .expect(expectedResponsecode);
+
+    return body;
+  }
+
+  const testDataList = [
+    {
+      testTitle: 'as ML Poweruser',
+      jobIds: [SINGLE_METRIC_JOB_CONFIG.job_id, MULTI_METRIC_JOB_CONFIG.job_id],
+      user: USER.ML_POWERUSER,
+      requestBody: {
+        datafeedIds: [
+          `datafeed-${SINGLE_METRIC_JOB_CONFIG.job_id}`,
+          `datafeed-${MULTI_METRIC_JOB_CONFIG.job_id}`,
+        ],
+      },
+      expected: {
+        responseCode: 200,
+        responseBody: {
+          [`datafeed-${SINGLE_METRIC_JOB_CONFIG.job_id}`]: { stopped: true },
+          [`datafeed-${MULTI_METRIC_JOB_CONFIG.job_id}`]: { stopped: true },
+        },
+      },
+    },
+  ];
+
+  const invalidTestDataList = [
+    {
+      testTitle: 'as ML Poweruser with datafeed ID that does not exist',
+      jobIds: [SINGLE_METRIC_JOB_CONFIG.job_id],
+      user: USER.ML_POWERUSER,
+      requestBody: {
+        datafeedIds: [`invalid-datafeed-${SINGLE_METRIC_JOB_CONFIG.job_id}`],
+      },
+      expected: {
+        responseCode: 200,
+        responseBody: {
+          [`invalid-datafeed-${SINGLE_METRIC_JOB_CONFIG.job_id}`]: { stopped: false },
+        },
+      },
+    },
+  ];
+
+  const testDataListUnauthorized = [
+    {
+      testTitle: 'as ML Unauthorized user',
+      user: USER.ML_UNAUTHORIZED,
+      jobIds: [SINGLE_METRIC_JOB_CONFIG.job_id, MULTI_METRIC_JOB_CONFIG.job_id],
+      requestBody: {
+        datafeedIds: [
+          `datafeed-${SINGLE_METRIC_JOB_CONFIG.job_id}`,
+          `datafeed-${MULTI_METRIC_JOB_CONFIG.job_id}`,
+        ],
+      },
+      expected: {
+        responseCode: 403,
+        error: 'Forbidden',
+      },
+    },
+    {
+      testTitle: 'as ML Viewer',
+      user: USER.ML_VIEWER,
+      jobIds: [SINGLE_METRIC_JOB_CONFIG.job_id, MULTI_METRIC_JOB_CONFIG.job_id],
+      requestBody: {
+        datafeedIds: [
+          `datafeed-${SINGLE_METRIC_JOB_CONFIG.job_id}`,
+          `datafeed-${MULTI_METRIC_JOB_CONFIG.job_id}`,
+        ],
+      },
+      expected: {
+        responseCode: 403,
+        error: 'Forbidden',
+      },
+    },
+  ];
+
+  describe('stop_datafeeds', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('sets up jobs', async () => {
+      for (const job of testSetupJobConfigs) {
+        const datafeedId = `datafeed-${job.job_id}`;
+        await ml.api.createAnomalyDetectionJob(job);
+        await ml.api.openAnomalyDetectionJob(job.job_id);
+        await ml.api.createDatafeed({
+          ...DATAFEED_CONFIG,
+          datafeed_id: datafeedId,
+          job_id: job.job_id,
+        });
+        await ml.api.startDatafeed(datafeedId, { start: '0' });
+        await ml.api.waitForDatafeedState(datafeedId, DATAFEED_STATE.STARTED);
+      }
+    });
```

Comment:
probably should be part of the `before` callback

Reply:
Done in https://github.com/elastic/kibana/pull/110961/commits/fa88781319a27a3ce24a458ed4745831f282b81e